crates/tako/src/server/monitoring.rs

```diff
@@ -0,0 +1,66 @@
+use crate::messages::gateway::{CollectedOverview, LostWorkerReason};
+use crate::messages::worker::WorkerOverview;
+use crate::WorkerId;
+
+use crate::messages::common::WorkerConfiguration;
+use serde::{Deserialize, Serialize};
+use std::time::SystemTime;
+
+pub struct EventHistory {
+    event_queue: Vec<MonitoringEvent>,
+    last_event_id: u32,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct MonitoringEvent {
+    pub message: MonitoringEventPayload,
+    pub id: u32,
+    pub time: SystemTime,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub enum MonitoringEventPayload {
+    WorkerConnected(WorkerId, WorkerConfiguration),
+    WorkerLost(WorkerId, LostWorkerReason),
+    OverviewUpdate(CollectedOverview),
+}
+
+impl EventHistory {
+    pub fn get_events(&self) -> &Vec<MonitoringEvent> {
+        &self.event_queue
+    }
+
+    pub fn on_worker_added(&mut self, id: WorkerId, configuration: WorkerConfiguration) {
+        self.event_queue.push(MonitoringEvent {
```

Comment:
`self.event_queue.push`, assigning the `id` and `time` is duplicated in all `on_...` methods. Please create a method for this, which will only receive the payload.